src/engraving/rendering/dev/tlayout.cpp

```diff
@@ -5922,6 +5922,10 @@ void TLayout::layoutTextLineBaseSegment(TextLineBaseSegment* item, LayoutContext
     // set end text position and extend bbox
     if (!item->endText()->empty()) {
         item->endText()->mutldata()->moveX(ldata->bbox().right());
+        //center end text for pedals (rosette star), prevents collision of rosette and "Ped" on following note
+        if (item->isPedalSegment() && !toPedalSegment(item)->pedal()->lineVisible()) {
+            item->endText()->mutldata()->moveX(-item->endText()->width() / 2);
+        }
```

Comment:
Shouldn't the star rather be right-aligned, i.e. so that the rightmost point of the star matches with the rightmost point of the hook if there were a hook?

Maybe a nicer way to achieve this would be to play with the `Pid::END_TEXT_ALIGN` property, but I haven't tested that myself yet. (See also `Pedal::propertyDefault` and the constructor of `Pedal`)

Reply:
Right-aligning is a great idea for the situation I fabricated, but with further research I have found that it should never arise when using these symbols.. if the pedal should be lifted and replaced on one note, the star is just omitted.

Admittedly I don't fully understand the whole style system, but from what I've tried ("just" adding a new entry to the enum in `styledef.h` and a value in `styledef.cpp`) it is very difficult to add a new attribute, in this case pedalEndTextAlign, to the style of the pedal. It looks like every entry in `textStyle.cpp` would have to be lengthened in order to accommodate different styling for each text in a textline, including some entries (like title text, with only one text item) with fully different property sets. This is because there is a set length for the list of properties a text style can have.

Making that length variable looks like a good option, I think. Everywhere else in the code, each text object on a textline is addressable individually, and it should be no different for how they're styled. That would be a much larger change, certainly outside of the scope of this bug, but do tell me if you think it's a good idea (pr not).

Reply:
I haven't had much time to work on this for the past couple weeks, and I don't really expect to in the future... I don't think the current solution is too bad since implementing it "properly" would require large changes to the code. From a design standpoint, I think it's more natural to have the notehead and asterisk centered on the same line (though the difference between this and right-justified is unnoticeably minor). Either way, I'd like to have the issue this PR was meant to address solved in the codebase, and would to see it closed, with or without commit 633ac7c. Having open PRs laying around seems a little offputting both to me and people starting work on new bugs.